src/components/util/Rivet.js

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright (C) 2018 The Trustees of Indiana University
+SPDX-License-Identifier: BSD-3-Clause
+*/
+import classNames from 'classnames'
+import React from 'react';
+// import PropTypes from 'prop-types';
+
+// Classes, Interfaces, and Types
+
+// export type Typescale = "base" | "xxs" | "xs" | "sm" | "md" | "lg" | "xl" | "xxl" | 12 | 14 | 16 | 18 | 20 | 23 | 26 | 29 | 32 | 36 | 41 | 46 | 52;
+// export type Size = 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';
+export const Sizes = [ 'xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl' ];
+// export type Edge = 'top' | 'right' | 'bottom' | 'left' | 'tb' | 'lr';
+export const Edges = [ 'top', 'right', 'bottom', 'left', 'tb', 'lr' ];
+// export type Display = "inline" | "inline-block" | "block" | "flex" | "flex-vertical-center";
+// export type Border = "all" | "radius" | Edge | Edge[]
+// export type Hidden = "sm-down" | "md-down" | "lg-down" | "xl-down" | "xxl-down" | "sm-up" | "md-up" | "lg-up" | "xl-up" | "xxl-up"
+
+// export interface Props {
+//     /**
+//      * Optional Rivet style: type scale adjustment
+//      * See: https://rivet.uits.iu.edu/components/layout/typography/
+//      */
+//     typescale?: Typescale,
+//     /**
+//      * Optional Rivet style: show/hide content based on screen size.
+//      * ex: 'md-down' will hide content on medium-sized screens and smaller.
+//      * ex: 'lg-up' will hide content on large-sized screens and larger.
+//      * See: https://rivet.uits.iu.edu/components/utilities/visibility/
+//      */
+//     hide?: Hidden,
+//     /**
+//      * Optional Rivet style: add a border to any or all sides of an element
+//      * See: https://rivet.uits.iu.edu/components/utilities/border/
+//      */
+//     border?: Border,
+//     /**
+//      * Optional Rivet style: adjust the margin(s) of an element
+//      * See: https://rivet.uits.iu.edu/components/layout/spacing/
+//      */
+//     margin?: Size | BoxStyle,
+//     /**
+//      * Optional Rivet style: adjust the padding(s) of an element
+//      * See: https://rivet.uits.iu.edu/components/layout/spacing/
+//      */
+//     padding?: Size | BoxStyle,
+//     /**
+//      * Optional Rivet style: adjust the display property of an element
+//      * See: https://rivet.uits.iu.edu/components/utilities/display/#display-property-utilities
+//      */
+//     display?: Display,
+// }
+
+// /**
+//  * An interface for describing the styling of box edges. 
+//  */
+// export interface BoxStyle {
+//     top?: Size,
+//     right?: Size,
+//     bottom?: Size,
+//     left?: Size,
+//     lr?: Size,
+//     tb?: Size,
+//     //
+//     xxs?: Edge | Edge[],
+//     xs?:  Edge | Edge[],
+//     sm?: Edge | Edge[],
+//     md?: Edge | Edge[],
+//     lg?: Edge | Edge[],
+//     xl?: Edge | Edge[],
+//     xxl?: Edge | Edge[],
+// }
+
```

Comment:
Please remove commented out code

Reply:
This code has been removed and validation moved to the appropriate functions.